systemd/systemd.go

```diff
@@ -941,24 +947,34 @@ loop:
 		case <-giveup.C:
 			break loop
 		case <-check.C:
-			bs, err := s.systemctl("show", "--property=ActiveState", serviceName)
-			if err != nil {
-				return err
+			allStopped := true
+			stillRunningServices := []string{}
+		serviceCheck:
+			for i := 0; i < len(serviceNames); i++ {
+				bs, err := s.systemctl("show", "--property=ActiveState", serviceNames[i])
+				if err != nil {
+					return err
+				}
+				if !isStopDone(bs) {
+					stillRunningServices = append(stillRunningServices, serviceNames[i])
+					allStopped = false
+				}
+				if !firstCheck {
+					continue serviceCheck
+				}
 			}
-			if isStopDone(bs) {
+			if allStopped {
 				return nil
 			}
-			if !firstCheck {
-				continue loop
-			}
+			serviceNames = stillRunningServices
 			firstCheck = false
 		case <-notify.C:
 		}
 		// after notify delay or after a failed first check
-		s.reporter.Notify(fmt.Sprintf("Waiting for %s to stop.", serviceName))
+		s.reporter.Notify(fmt.Sprintf("Waiting for %q to stop.", serviceNames))
 	}
 
-	return &Timeout{action: "stop", service: serviceName}
+	return &Timeout{action: "stop", service: serviceNames[0]}
```

Comment:
Timeout could be updated to have a slice of service names instead and we could set them here so that they're all included in the err message

Reply:
Yes I was thinking to do this in separate PR, but may be better to include it here.
Where is `Timeout` defined, to change it to the slice of services?

Reply:
This is done now.